account_move_update_analytic/wizards/account_move_update_analytic.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 Le Filament
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMoveUpdateAnalytic(models.TransientModel):
+    _name = "account.move.update.analytic.wizard"
+    _description = "Account Move Update Analytic Account Wizard"
+
+    line_id = fields.Many2one("account.move.line", string="Invoice line")
+    current_analytic_account_id = fields.Many2one(
+        related="line_id.analytic_account_id", string="Current Analytic Account"
+    )
+    current_analytic_tag_ids = fields.Many2many(
+        related="line_id.analytic_tag_ids", string="Current Analytic Tags"
+    )
+    company_id = fields.Many2one(related="line_id.company_id")
+    new_analytic_account_id = fields.Many2one(
+        "account.analytic.account", string="New Analytic Account", check_company=True
+    )
+    new_analytic_tag_ids = fields.Many2many(
+        "account.analytic.tag", string="New Analytic Tags", check_company=True
+    )
+
+    @api.model
+    def default_get(self, fields):
+        rec = super(AccountMoveUpdateAnalytic, self).default_get(fields)
+        context = dict(self._context or {})
+        active_id = context.get("active_id", False)
+        aml = self.env["account.move.line"].browse(active_id)
+        rec.update(
+            {
+                "line_id": active_id,
+                "current_analytic_account_id": aml.analytic_account_id.id,
+                "current_analytic_tag_ids": [(6, 0, aml.analytic_tag_ids.ids or [])],
+                "company_id": aml.company_id.id,
+            }
+        )
+        return rec
+
+    def update_analytic_lines(self):
+        self.line_id.analytic_line_ids.unlink()
+        if self.user_has_groups("analytic.group_analytic_accounting"):
+            self.line_id.write(
+                {"analytic_account_id": self.new_analytic_account_id.id or False}
+            )
+        if self.user_has_groups("analytic.group_analytic_tags"):
+            self.line_id.write(
```

Comment:
Same as above and not need last `[]` because .ids always return `[]` if it's an empty recordset.
I'm not 100% sure if the `[(6, 0, ...)]` it's needed in this case, but trying it will be great.
```suggestion
            self.line_id.analytic_tag_ids = self.new_analytic_tag_ids.ids
```

Reply:
I removed the or [] but had to keep the [(6,0,xx)] otherwise it does not empty the field if you remove analytic tags